pkg/controllers/proxy/network_services_controller.go

```diff
@@ -2027,32 +2016,108 @@ func (nsc *NetworkServicesController) Cleanup() {
 func (nsc *NetworkServicesController) newEndpointsEventHandler() cache.ResourceEventHandler {
 	return cache.ResourceEventHandlerFuncs{
 		AddFunc: func(obj interface{}) {
-			nsc.OnEndpointsUpdate(obj)
+			nsc.handleEndpointsAdd(obj)
 
 		},
 		UpdateFunc: func(oldObj, newObj interface{}) {
-			nsc.OnEndpointsUpdate(newObj)
+			nsc.handleEndpointsUpdate(oldObj, newObj)
 		},
 		DeleteFunc: func(obj interface{}) {
-			nsc.OnEndpointsUpdate(obj)
-
+			nsc.handleEndpointsDelete(obj)
 		},
 	}
 }
 
+func (nsc *NetworkServicesController) handleEndpointsAdd(obj interface{}) {
+	endpoints, ok := obj.(*api.Endpoints)
+	if !ok {
+		glog.Errorf("unexpected object type: %v", obj)
+		return
+	}
+	nsc.OnEndpointsUpdate(endpoints)
+}
+
+func (nsc *NetworkServicesController) handleEndpointsUpdate(oldObj, newObj interface{}) {
+	_, ok := oldObj.(*api.Endpoints)
```

Comment:
Is there a specific reason to process these at all if we don't use them? While I would expect that we would ever get this error, I would imagine that the greater benefit for the stability of the controller would be to process the update anyway assuming that we could cast the newEndpoints and send them on to the OnUpdate function.

Reply:
Ideally we should use both the new and old objects to check if there is an update which deserved processing rule.

Somthing like https://github.com/cloudnativelabs/kube-router/blob/v1.0.0-rc1/pkg/controllers/netpol/network_policy_controller.go#L1628

I will raise a seperate PR to address this issue.